src/lib/getEventType.ts

```diff
@@ -0,0 +1,192 @@
+import { ParsedQueryString } from "./models";
+
+type BrokenEvent = "Broken Event";
+type PingType =
+  | "HeartBeat"
+  | "Deanonymization"
+  | "Page performance metric"
+  | "Custom"
+  | BrokenEvent;
+type EventType =
+  | "Goal Conversion"
+  | "Ping"
+  | "Download"
+  | "Outlink"
+  | "Consent form impression"
+  | "Consent form click"
+  | "Consent decision"
+  | "SharePoint"
+  | "Custom event"
+  | "Content interaction"
+  | "Content impression"
+  | "Cart update"
+  | "Product detail view"
+  | "Add to cart"
+  | "Remove from cart"
+  | "Order completed"
+  | "Internal search"
+  | "Page view"
+  | PingType
+  | BrokenEvent;
+/**
+ * https://help.piwik.pro/support/questions/what-are-events-and-how-are-they-detected/
+ */
+export function getEventType(eventParams: ParsedQueryString[]): EventType {
+  if (isGoalConversion(eventParams)) return "Goal Conversion";
+
+  if (isPing(eventParams)) return getPingType(eventParams);
+
+  if (isDownload(eventParams)) return "Download";
+
+  if (isOutlink(eventParams)) return "Outlink";
+
+  if (isConsentFormImpression(eventParams)) return "Consent form impression";
+
+  if (isConsentFormClick(eventParams)) return "Consent form click";
+
+  if (isConsentDecision(eventParams)) return "Consent decision";
+
+  if (isSharePoint(eventParams)) return "SharePoint";
```

Comment:
`JSON.parse` inside `isSharePoint` could potentially throw an error, we could catch it here and report invalid paylod structure error

Reply:
Something like this?

```
function isSharePoint(eventParams: ParsedQueryString[]) {
  const eventCustomVariables = eventParams.find((p) => p.name === "cvar");
  const sessionCustomVariables = eventParams.find((p) => p.name === "_cvar");
  const category = eventParams.find((p) => p.name === "e_c");

  try {
    if (eventCustomVariables) {
      const parsedEventCustomVariables = JSON.parse(decodeURIComponent(eventCustomVariables.value));
    }
  } catch (e) {
    console.error("Error parsing event custom variables:", e);
    return false;
  }

  try {
    if (sessionCustomVariables) {
      const parsedSessionCustomVariables = JSON.parse(decodeURIComponent(sessionCustomVariables.value));
    }
  } catch (e) {
    console.error("Error parsing session custom variables:", e);
    return false;
  }

  return (
    (parsedEventCustomVariables?.["1"]?.[0] === "ppas.sharepoint.plugin" ||
      parsedSessionCustomVariables?.["1"]?.[0] === "ppas.sharepoint.plugin") &&
    (category?.value === "download" || category?.value === "search")
  );
}
```

Reply:
The simplest solution would be to catch the error at the top level
```ts
try{
    if (isSharePoint(eventParams)) return "SharePoint";
}
catch(e){
    console.log(e)
    alert("error while parsing custom variables ('cvar' or '_cvar' parameter)")
    return "Broken Event"
}
```
More fine grained errors are also cool, I don't have strong opinion on that